flaky test TestDoImport

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
 === RUN   TestDoImport
    local_test.go:2048: 
        	Error Trace:	br/pkg/lightning/backend/local/local_test.go:2048
        	Error:      	Error "context canceled" does not contain "meet error when generateJobForRange again"
        	Test:       	TestDoImport
--- FAIL: TestDoImport (1.01s) 
```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/ingestor/ingestctrl/local.go
+++ b/pkg/ingestor/ingestctrl/local.go
@@ -1692,7 +1692,9 @@ func (local *Backend) doImport(
 		// Close the pool, as well as the channel.
 		wctx.Cancel()
 		pool.Release()
-		return nil
+		// The worker may set operator error after jobWg.Wait() is unblocked. Re-check
+		// here to avoid losing worker errors due to cancellation ordering.
+		return wctx.OperatorErr()
 	})
 
 	err := workGroup.Wait()

